scipy/stats/_morestats.py

```diff
@@ -3133,8 +3133,11 @@ def wilcoxon(x, y=None, zero_method="wilcox", correction=False,
             raise ValueError('The samples x and y must have the same length.')
         d = x - y
 
+    if len(d) == 0:
+        return WilcoxonResult(np.nan, np.nan)
+
     if mode == "auto":
-        if len(d) <= 25:
+        if len(d) <= 50:
```

Comment:
Let's also consider whether there are ties when choosing the method. See gh-13690.